dask_geopandas/expr.py

```diff
@@ -115,7 +115,7 @@ def __init__(self, expr, spatial_partitions=None):
         super().__init__(expr)
         self.spatial_partitions = spatial_partitions
 
-    def to_legacy_dataframe(self, optimize: bool = True, **optimize_kwargs):
+    def to_dask_dataframe(self):
```

Comment:
@TomAugspurger I renamed this back to the original name, but now this is fine because dask-expr upstream renamed the `to_dask_dataframe` they added to `to_legacy_dataframe`, so there is no longer a name clash.

`to_legacy_dataframe` will convert to a legacy dask.dataframe collection, which is needed for certain implementations that still use the original implementation (e.g. dask still does this for parquet IO right now). 

While our `to_dask_dataframe` is meant to convert your dask-geopandas object to a dask object (regardless of it being a legacy collection or a new expression). 

(and so naming this implementation `to_legacy_dataframe` actually broke the parquet tests, because it is not doing what dask is expecting, i.e. it doesn't return a legacy collection, just the same object but where the partitions are pd.DataFrames instead of GeoDataFrames)

Reply:
Ah, makes sense, thanks.